sky/provision/aws/instance.py

```diff
@@ -164,26 +164,29 @@ def _create_instances(ec2_fail_fast, cluster_name: str, node_config: Dict[str,
         'MaxCount': count,
         'TagSpecifications': tag_specs
     })
-
+    # We are adding 'NetworkInterfaces' in the inner loop and having both keys
+    # is considered invalid by the create_instances API.
+    security_group_ids = conf.pop('SecurityGroupIds', None)
```

Comment:
Is there a case the `security_group_ids` will be `None` in our implementation? Will that cause an failure in the `NetworkInterface`?

Reply:
Good catch - using None here would make the request throw TypeError. That said, the bootstrapping phase guarantees to create-or-get an SG:

https://github.com/skypilot-org/skypilot/blob/23aa6dd3f6daf49544d4ddd68acbce422c9d3d03/sky/provision/aws/config.py#L94-L96

Added an assert here. PTAL.